lib/ecto_job/producer.ex

```diff
@@ -24,16 +24,17 @@ defmodule EctoJob.Producer do
     @moduledoc """
     Internal state of the Producer GenStage
     """
-    @enforce_keys [:repo, :schema, :notifier, :demand, :clock, :poll_interval]
-    defstruct [:repo, :schema, :notifier, :demand, :clock, :poll_interval]
+    @enforce_keys [:repo, :schema, :notifier, :demand, :clock, :poll_interval, :base_expiry_seconds]
+    defstruct [:repo, :schema, :notifier, :demand, :clock, :poll_interval, :base_expiry_seconds]
 
     @type t :: %__MODULE__{
             repo: EctoJob.Producer.repo(),
             schema: EctoJob.Producer.schema(),
             notifier: EctoJob.Producer.notifier(),
             demand: integer,
             clock: (() -> DateTime.t()),
-            poll_interval: non_neg_integer()
+            poll_interval: non_neg_integer(),
+            base_expiry_seconds: non_neg_integer()
```

Comment:
For someone unfamiliar with the code, I wonder if a name like `execution_timeout` would be more obvious?
I'm tempted to also configure `reservation_timeout` separately, since it is safer to set to a lower value.

Reply:
Those both sound like fine changes to me. I'll have a change to edit in a couple days

Reply:
👍 let's use ms